### PR TITLE
PR - adding user error callback to form

### PR DESCRIPTION
**Pull request by JulienParis**
There was no indication adressing the user if the form was not correctly sent to the backend (except the alert border being red). 

Here comes an light solution to this, mainly putting a custom text for this case and inviting the user to contact us by mail => Solves #136 
 
Capture d’écran 2023-10-02 à 16 16 06

Preview for tests on `staging` : https://tee-frontend-staging.osc-fr1.scalingo.io/

### CODE REVIEW

**Comment by JulienParis** [comment]
note: depending on how we want to reshape the `.env` variables in the future (in particular `VITE_BREVO_SENDER_EMAIL`), let's keep in mind the following `ref` in `TeeForm.ts` might change : 

```js
const contactEmail = ref<string>(metaEnv.VITE_BREVO_SENDER_EMAIL || 'france-transition@beta.gouv.fr')
```

**Comment by yvalentin** [comment]
> note: @yvalentin depending on how we want to reshape the `.env` variables in the future (in particular `VITE_BREVO_SENDER_EMAIL`), let's keep in mind the following `ref` in `TeeForm.ts` might change :
> 
> ```js
> const contactEmail = ref<string>(metaEnv.VITE_BREVO_SENDER_EMAIL || 'france-transition@beta.gouv.fr')
> ```

Ok !

suggestion: Why we need a `ref` on `contactEmail`, it seems this don't need to be a dynamic attribute ?

File: packages/web/src/components/TeeForm.vue [hunk 1]
```diff
@@ -226,6 +250,10 @@ const analytics = analyticsStore()
 // const usedTracks: UsedTrack[] | any[] = tracks.getAllUsedTracks
 const trackValues: any[] = tracks.getAllUsedTracksValues
 
+// @ts-ignore
+const metaEnv = import.meta.env
+const contactEmail = ref<string>(metaEnv.VITE_BREVO_SENDER_EMAIL || 'france-transition@beta.gouv.fr')
```
Comment (yvalentin) [inline packages/web/src/components/TeeForm.vue:255]: suggestion: Why we need a `ref` on `contactEmail`, it seems this don't need to be a dynamic attribute ?

**Review by yvalentin — CHANGES_REQUESTED** [review]
suggestion: On `request.ts` from `utils` folder, on the `sendRequest` method who call the backend api. What's happen if the serveur is down or other errors like timeout. Shoud we add a `try/catch` arround the `fetch` request to display a generic error message to user

File: packages/web/src/components/TeeForm.vue (hunk 1 above)
Comment (JulienParis) [reply packages/web/src/components/TeeForm.vue:255]: Note : I think at the time I needed to display this into the `debug` spaces into the html template. You're right it doesn't need to be reactive

Chore: (for me) I should remove the `ref<string>` occurence

Comment (JulienParis) [inline packages/web/src/components/TeeForm.vue:255]: @yvalentin 

suggestion: to replace existing line, WDYT ?

```suggestion
const contactEmail = metaEnv.VITE_BREVO_SENDER_EMAIL || 'france-transition@beta.gouv.fr'
```

Comment (yvalentin) [reply packages/web/src/components/TeeForm.vue:255]: yes but can you type the constant `contactEmail`

```js
const contactEmail: string = metaEnv.VITE_BREVO_SENDER_EMAIL || 'france-transition@beta.gouv.fr'
```

Comment (yvalentin) [reply packages/web/src/components/TeeForm.vue:255]: suggestion: @JulienParis And maybe, we need to change the name of `VITE_BREVO_SENDER_EMAIL` to something like: `VITE_CONTACT_EMAIL` ?

**Comment by JulienParis** [comment]
Notes : 

Missing feats to develop before merging this PR 

- [ ] add `try / catch`
- [ ] add loader to signify request is sent to user

File: packages/web/src/components/TeeForm.vue (hunk 1 above)
Comment (JulienParis) [reply packages/web/src/components/TeeForm.vue]: note: changed in [this commit](https://github.com/betagouv/transition-ecologique-entreprises-widget/pull/135/commits/3df4c62f34668a1ec06d8c70fd3b19ed8acf664e)

Comment (JulienParis) [reply packages/web/src/components/TeeForm.vue]: note: `ref<string>` use changed in [this commit](https://github.com/betagouv/transition-ecologique-entreprises-widget/pull/135/commits/3df4c62f34668a1ec06d8c70fd3b19ed8acf664e)

File: packages/web/src/utils/requests.ts [hunk 2]
```diff
@@ -68,20 +68,34 @@ export const sendRequest = async (url: string, method: string, headers: any, bod
...
+    // console.log('utils > requests > sendRequest >  respJson :', respJson)
+    return respJson
+  } catch (error) {
+    const respObj: ReqResp = {}
+    console.log('utils > requests > sendRequest >  error :', error)
+    respObj.action = action
+    respObj.ok = false
+    respObj.status = 404
+    respObj.statusText = 'Error'
+    respObj.message = `${error}`
```
Comment (yvalentin) [inline packages/web/src/utils/requests.ts:96]: suggestion: A mon avis, ici on est plus dans le cas d'un 500 ou 408 (timeout) que une 404.  
Que penses-tu de mettre un status code 500 avec un message générique 'Erreur interne du serveur' ?

PS: Le console log devrait être commenté, non ?

Comment (JulienParis) [reply packages/web/src/utils/requests.ts]: praise: thanks, I made these changes [here](https://github.com/betagouv/transition-ecologique-entreprises-widget/pull/135/commits/86ee3fbdb15408712bd8ce000fe5b9e150715fa7)

File: packages/web/src/components/TeeForm.vue (hunk 1 above)
Comment (JulienParis) [reply packages/web/src/components/TeeForm.vue]: note: as I changed the target branch and re-synchronised locally my branch `j-form-error-msg` with `preprod`, this change is now included. I can mark this conversation as resolved

**Review by JulienParis — COMMENTED** [review]
@yvalentin : requested changes seems integrated now. Can you have a look to approve (or not) ?

**Review by yvalentin — APPROVED** [review]

**Pull request merged**